std/msgfilter_test.go

```diff
@@ -159,44 +348,80 @@ func TestMsgFilter_SetPrefix(t *testing.T) {
 }
 
 func TestMsgFilter_HasPrefix(t *testing.T) {
+	messageCreate := func(content string) interface{} {
+		return &disgord.MessageCreate{
+			Message: &disgord.Message{Content: content},
+		}
+	}
+
+	messageUpdate := func(content string) interface{} {
+		return &disgord.MessageUpdate{
+			Message: &disgord.Message{Content: content},
+		}
+	}
+
 	prefix := "!!"
+	testCases := []struct {
+		name              string
+		evt               interface{}
+		shouldPassThrough bool
+	}{
+		{"MessageCreate_CorrectPrefix", messageCreate(prefix + "hello"), true},
+		{"MessageUpdate_CorrectPrefix", messageUpdate(prefix + "hello"), true},
+		{"MessageCreate_WrongPrefix", messageCreate("diff prefix " + prefix + "hello"), false},
+		{"MessageUpdate_WrongPrefix", messageUpdate("diff prefix " + prefix + "hello"), false},
+	}
+
 	filter, _ := newMsgFilter(context.Background(), &clientRESTMock{})
 	filter.SetPrefix(prefix)
 
-	var evt interface{}
-	e := &disgord.MessageCreate{
-		Message: &disgord.Message{Content: prefix + "hello"},
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := filter.HasPrefix(tc.evt)
+			if tc.shouldPassThrough && result == nil {
+				t.Error("expected to passthrough")
+			}
+			if !tc.shouldPassThrough && result != nil {
+				t.Error("expected a filter match")
+			}
+		})
 	}
-	evt = e
+}
 
-	result := filter.HasPrefix(evt)
-	if result == nil {
-		t.Error("expected to find a match")
+func TestMsgFilter_StripPrefix(t *testing.T) {
+	messageCreate := func(content string) interface{} {
+		return &disgord.MessageCreate{
+			Message: &disgord.Message{Content: content},
+		}
 	}
 
-	e.Message.Content = "diff prefix " + e.Message.Content
-	result = filter.HasBotMentionPrefix(evt)
```

Comment:
this is intentionally changed, looks like this was a bug in the test (this test is testing HasPrefix)